src/GraphQL.Tests/Instrumentation/OpenTelemetryTests.cs

```diff
@@ -0,0 +1,245 @@
+#if NET5_0_OR_GREATER
+
+using System.Diagnostics;
+using GraphQL.Instrumentation;
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+
+namespace GraphQL.Tests.Instrumentation;
+
+public class OpenTelemetryTests : IDisposable
+{
+    private readonly List<Activity> _exportedActivities = new();
+    private readonly IHost _host;
+    private readonly IDocumentExecuter<ISchema> _executer;
+    private readonly IGraphQLTextSerializer _serializer;
+
+    public OpenTelemetryTests()
+    {
+        // configure services
+        _host = new HostBuilder()
+            .ConfigureServices(services =>
+            {
+                services.AddOpenTelemetry()
+                    .WithTracing(b => b
+                        .AddSource(GraphQLTelemetryProvider.SourceName) // need to specify the source name to be traced
+                        .AddInMemoryExporter(_exportedActivities))
+                    .StartWithHost(); // relies on HostBuilder to start telemetry services (typical for ASP.NET applications)
```

Comment:
```suggestion
                        .AddInMemoryExporter(_exportedActivities));
```

From MS sources:
```
  public static OpenTelemetryBuilder AddOpenTelemetry(this IServiceCollection services)
    {
        Guard.ThrowIfNull(services);

        services.TryAddEnumerable(
            ServiceDescriptor.Singleton<IHostedService, TelemetryHostedService>());

        return new(services);
    }

public static OpenTelemetryBuilder StartWithHost(this OpenTelemetryBuilder builder)
    {
        Guard.ThrowIfNull(builder);

        builder.Services.TryAddEnumerable(
            ServiceDescriptor.Singleton<IHostedService, TelemetryHostedService>());

        return builder;
    }
```

Reply:
I'll be happy to remove the line of code if this is recommended by MS.  On the other hand, if MS simply does not have any specific code to add within `StartWithHost` at present, then the code is correct as written, so it will work properly in future versions of .NET.  We should not be looking at the source code to determine this.

Reply:
The `StartWithHost` was removed in rc4 which was [released](https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/core-1.4.0-rc.4) yesterday

Reply:
👍

Reply:
😄

Reply:
It was late feedback on the 1.4.0 API changes. Sorry for that :)